### PR TITLE
Adding in progress overlay/state for wallet recovery

### LINKED ISSUE #11790: loading/progress indicator for wallet recovery
### Description
Several users have mentioned that after entering or importing their BAT wallet recovery keys, it doesn't appear as though anything is happening on the UI. They have requested some kind of loading or progress indicator to let the user know something is happening. If a progress indicator isn't possible, maybe some kind of transition overlay (spinning circle?) to let the user know something is going on in the background.

### Steps to Reproduce

  1. Have backup words for a BAT wallet.
  2. Create a new profile and enable payments.
  3. Recover your wallet.

**Actual result:**
Sometimes wallet recovery takes up to 10 minutes, during this time there is no visual indication in the UI that anything is happening.

**Expected result:**
Some kind of visual notifier that a wallet recovery is being attempted.

**Reproduces how often:**
unknown. sometimes wallets recover fast, so this isn't an issue. other times it takes up to 10 minutes.

### Brave Version
BAT Mercury enabled versions (0.19.48, 0.19.70, etc)

**Reproducible on current live release:**
yes

### PR DESCRIPTION
**Pull request by ryanml**
Fixes #11790

With [1ef48b8](https://github.com/ryanml/browser-laptop/commit/1ef48b82f46a19240e24f2f8296d703d41cf4ff8) an overlay is added to the recovery dialogue while the wallet recovery is in progress. Included in the overlay is a 'Cancel' button which will cancel the current recovery while it is in progress.

Here is the PR in action:
http://recordit.co/OYC6OgUjMX

**Note: An artificial delay was introduced in to the backup for display purposes, in many instances the recovery is quite fast**

~~This PR is currently marked as WIP. @bradleyrichter had an idea for a pretty slick implementation of text animation with a minimum propagation time. Example:~~

~~The Recovery In Progress Overlay may include timed progress steps like:~~

~~Reading recovery file….~~
~~Processing wallet…~~
~~Wallet Recovered! Go spend it!~~

~~This will involve some refactoring/fail cases in the recovery state, and will involve some additional work from my end and perhaps some more UX oriented discussion. Further work will be pushed right to this PR.~~

**We will be tackling this in a separate issue :)**

## Submitter Checklist:
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:
**For current in progress incarnation**

1. Start Brave with a clean profile, enable payments.
2. Exit Brave and corrupt your wallet by removing seeds 20 and 21 from ledger-state.json
3. Restart brave and go to `about:preferences#payments`
4. Recover wallet with a bad recovery key
5. Ensure that the in progress state is shown while the recovery attempts.
6. Recover the wallet with `vower obovoid menace tobogganist hoyle honoree pixel pestilently disconcertment sellable ruffing supervision zoroastrian based coparent slackened`
7. Ensure that the in progress state is shown while the recovery attempts.

Let me know if help is needed introducing a manual delay via the codebase for testing purposes.

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [x] Adequate test coverage exists to prevent regressions
- [x] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/common/state/ledgerState.js [hunk 1]
```diff
@@ -356,6 +356,7 @@ const ledgerState = {
   setRecoveryStatus: (state, status) => {
     state = validateState(state)
     const date = new Date().getTime()
+    state = state.setIn(['about', 'preferences', 'recoveryInProgress'], false)
```
Comment (jasonrsadler) [inline app/common/state/ledgerState.js]: should this be `state.setIn([....], !status)` or will it always be false on setRecoveryStatus?

Comment (ryanml) [reply app/common/state/ledgerState.js]: the `setRecoveryStatus` method inside ledger state is called when the conclusion of the recovery has occurred, (success/fail) or when it is reset. In both cases `recoveryInProgress` should be set to false as the attempted recovery is over.

**Comment by ryanml** [comment]
@jasonrsadler I will see what's going on with the cancellation 👍

File: app/common/state/ledgerState.js (hunk 1 above)
Comment (NejcZdovc) [reply app/common/state/ledgerState.js]: please use `aboutPreferencesState.setPreferencesProp` here as well

Comment (NejcZdovc) [reply app/common/state/ledgerState.js]: please move this function `setRecoveryStatus` into `aboutPreferencesState` and use `setPreferencesProp` for all 3 props. Thank you

Comment (ryanml) [reply app/common/state/ledgerState.js]: @NejcZdovc refactoring pushed

File: app/renderer/components/preferences/payment/ledgerRecovery.js [hunk 2]
```diff
@@ -58,12 +58,18 @@ class LedgerRecoveryContent extends ImmutableComponent {
     }
   }
 
+  onCancelRecovery (status) {
+    this.onRecoveryOverlay(status)
+    this.clearRecoveryStatus(status)
```
Comment (NejcZdovc) [inline app/renderer/components/preferences/payment/ledgerRecovery.js]: I think this call is not needed, because we already have it in `onRecoveryOverlay`

**Comment by ryanml** [comment]
squashed

**Comment by NejcZdovc** [comment]
master 417f3b7
0.23 61fedd8d9
0.22.x-release3 47a615e2e

**Comment by srirambv** [comment]
@ryanml  should the `Recovering progress` text show right away when you click on import recovery keys?

**Comment by ryanml** [comment]
@srirambv no it should not, it should only be showing when an actual recovery is in progress. Can you file issue? Thanks

**Pull request merged**